modules/assim.batch/R/pda.mcmc.R

```diff
@@ -1,40 +1,78 @@
-## Paramater Data Assimilation using MCMC
-## Mike Dietze
-##
-## Brute-force, only to be used on simple models
-##
-## model = name of model (string)
-## chain = ID number of mcmc chain
-## params = previous output (used when updating MCMC)
-## var.names = vector of names of which parameters in the parameter vector to vary
-## jvar = jump variance
-pda.mcmc <- function(settings,prior,chain=1,var.names=NULL,jvar=NULL,params=NULL){
+##' Paramater Data Assimilation using MCMC
+##'
+##' Brute-force, only to be used on simple models
+##'
+##' @title Paramater Data Assimilation using MCMC
+##' @param settings = a pecan settings list
+##'
+##' @return nothing. Diagnostic plots, MCMC samples, and posterior distributions
+##'  are saved as files and db records.
+##'
+##' @author Mike Dietze
+##' @author Ryan Kelly
+pda.mcmc <- function(settings, params=NULL, jvar=NULL, var.names=NULL, prior=NULL, chain=NULL,
+                     adapt=NULL, adj.min=NULL, ar.target=NULL){
+  # Quit if pda not requested in settings
+  if(!('assim.batch' %in% names(settings))) {
+    return()
+  }
+
+  require(coda)
   
   ## this bit of code is useful for defining the variables passed to this function 
```

Comment:
This was one of the toughest things I found about debugging R functions - having the objects you need inside the function in the form that was causing a bug.

A few solutions I've found:

* `debugonce(pda.mcmc)` before running the function will put you inside the function with all of the variables you need. This is the most important all-around workhorse
* but to get there you need an example with such inputs. These can go, e.g.
 * in tests 
 * in vignettes 
 * in function examples